packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -564,6 +579,35 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
+    private closestPoint(array: { x: number, y: number}[], chartWidth: number, chartHeight: number): any | undefined {
+        let min_hypotenuse = Number.MAX_VALUE;
+        let closestPoint = undefined;
+        const offset = this.props.viewRange.getOffset() ?? BigInt(0);
+        const start = this.props.viewRange.getStart();
+        const end = this.props.viewRange.getEnd();
+        const xRatio = chartWidth / Number(end - start);
+        const yRatio = (chartHeight - this.margin.top - this.margin.bottom) / (this.state.allMax - this.state.allMin);
+
+        array.forEach((ar: {x: number, y: number}) => {
+            const x = (ar.x - Number(start - offset)) * xRatio;
+            const y = (ar.y - this.state.allMin) * yRatio + this.margin.bottom;
+            const distX = this.positionXMove - x;
+            const distY = chartHeight - this.positionYMove - y;
+            const hypotenuse = distX * distX + distY * distY;
+
+            if (min_hypotenuse > hypotenuse){
+                closestPoint = ar;
+                min_hypotenuse = hypotenuse;
+            }
+        });
+
+        if (min_hypotenuse < 400) {
```

Comment:
comment explaining 20 pixels in a circle, so max 14 x and 14 y?

Reply:
To be clear, for future devs need to modify this, explain why 400 is picked.

Reply:
Done.